README.md

```diff
@@ -44,8 +44,9 @@ This project has a special dedication to Tony Morris' blog post [Debut with a ca
 I'm also very thankful to [@sviperll](https://github.com/sviperll) and his [adt4j](https://github.com/sviperll/adt4j/) project which was the initial inspiration for Derive4J.
 
 So. What can this project do for us, poor functional programmers stuck with a legacy language called Java?
-A good deal of what is available for free in better languages like Haskell: pattern matching, laziness...
-An example being worth a thousand words...
+A good deal of what is commonly available in better languages like Haskell: structural pattern matching, laziness, lenses and prisms.
+
+Algebraic data types come in two flavours, product types and sum types. This readme focus on sum types because it is the more interesting case; product types being the well known common case in java. But Derive4J handles product types in exactly the same fashion (ie. through a visitor interface with a single abstract method).
```

Comment:
It's good, though the new paragraph seems out of place under the list of features. Perhaps if it were closer to the previous mention of ADTs, or if ADTs were mentioned in the feature list, eg "[...] structural pattern matching (and/with/on) Algebraic Data Types, [...]"

Reply:
updated intro paragraph. tell me if further remarks.